### ISSUE TITLE
nx-release-publish fails after upgrading to 19.5.X

### ISSUE THREAD

**Issue by or-orchid** [nrwl/nx#27440]
### Current Behavior

After upgrading NX to 19.5.X, the publish command fails

```
>  nx release publish --dry-run --verbose
Based on your config, the following projects were matched for publishing but do not have the "nx-release-publish" target specified:

- eslint-plugin
- healthcheck
- common
- logger
- redis
- test

This is usually caused by not having an appropriate plugin, such as "@nx/js" installed, which will add the appropriate "nx-release-publish" target for you automatically.
```

### Expected Behavior

Should publish libs

### Steps to Reproduce

1. Install nx:19.4.3
2. Run ` nx release publish --dry-run --verbose` and see that the packages should be published
3. run `nx migrate 19.5.7`
4. run `nx migrate --run-migrations --if-exists`
5. Run ` nx release publish --dry-run --verbose` and see the errors

### Nx Report

```shell
➜  nodejs-core git:(main) ✗ nx report

 NX   Report complete - copy this into the issue template

Node           : 20.10.0
OS             : darwin-arm64
Native Target  : aarch64-macos
npm            : 10.8.2

nx                 : 19.5.7
@nx/js             : 19.5.7
@nx/jest           : 19.5.7
@nx/linter         : 19.5.7
@nx/eslint         : 19.5.7
@nx/workspace      : 19.5.7
@nx/devkit         : 19.5.7
@nx/eslint-plugin  : 19.5.7
@nx/nest           : 19.5.7
@nx/node           : 19.5.7
@nx/plugin         : 19.5.7
@nrwl/tao          : 19.5.7
typescript         : 5.5.4
---------------------------------------
Registered Plugins:
@nx/eslint/plugin
@nx/jest/plugin
---------------------------------------
Community plugins:
@nx-aws-plugin/nx-aws-cache : 3.2.1

```

### Failure Logs

```
This is usually caused by not having an appropriate plugin, such as "@nx/js" installed, which will add the appropriate "nx-release-publish" target for you automatically.

    at runPublishOnProjects (/Users/orgaizer/repos/nodejs-core/node_modules/nx/src/command-line/release/publish.js:127:15)
    at releasePublish (/Users/orgaizer/repos/nodejs-core/node_modules/nx/src/command-line/release/publish.js:72:30)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at handleErrors (/Users/orgaizer/repos/nodejs-core/node_modules/nx/src/utils/params.js:22:24)
    at Object.handler (/Users/orgaizer/repos/nodejs-core/node_modules/nx/src/command-line/release/command-object.js:185:24)
```

### Package Manager Version

10.8.2

### Operating System

- [X] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

I suspect it's related to this [PR](https://github.com/nrwl/nx/pull/26596)

**Comment by JamesHenry** [comment]
Please share your `nx.json` contents and any `"release"` or `"nx-release-publish"` config found in `project.json` files

**Comment by or-orchid** [comment]
@JamesHenry Hey,attached
```
{
  "pluginsConfig": {
    "@nx/js": {
      "analyzeSourceFiles": true
    }
  },
  "$schema": "./node_modules/nx/schemas/nx-schema.json",
  "extends": "nx/presets/npm.json",
  "namedInputs": {
    "default": ["{projectRoot}/**/*", "sharedGlobals"],
    "production": [
      "default",
      "!{projectRoot}/.eslintrc.json",
      "!{projectRoot}/eslint.config.js",
      "!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap)",
      "!{projectRoot}/tsconfig.spec.json",
      "!{projectRoot}/jest.config.[jt]s",
      "!{projectRoot}/src/test-setup.[jt]s",
      "!{projectRoot}/test-setup.[jt]s"
    ],
    "sharedGlobals": []
  },
  "defaultBase": "main",
  "plugins": [
    {
      "plugin": "@nx/eslint/plugin",
      "options": {
        "targetName": "lint"
      }
    },
    {
      "plugin": "@nx/jest/plugin",
      "options": {
        "targetName": "test"
      },
      "exclude": ["test-e2e/**/*", "apps/test-e2e/**/*"]
    },
    {
      "plugin": "@nx/webpack/plugin",
      "options": {
        "buildTargetName": "build",
        "serveTargetName": "serve",
        "previewTargetName": "preview"
      }
    }
  ],
  "targetDefaults": {
    "@nx/js:tsc": {
      "cache": true,
      "dependsOn": ["^build"],
      "inputs": ["production", "^production"],
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/{projectRoot}",
        "tsConfig": "{projectRoot}/tsconfig.lib.json",
        "packageJson": "{projectRoot}/package.json",
        "main": "{projectRoot}/src/index.ts"
      }
    },
    "@nx/js:eslint": {
      "cache": true,
      "dependsOn": ["^build"],
      "inputs": ["default", "^default"],
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "{projectRoot}/**/*.{ts,tsx,js,jsx}",
          "{projectRoot}/package.json"
        ]
      }
    },
    "@nx/jest:jest": {
      "cache": true,
      "dependsOn": ["^build"],
      "inputs": ["default", "^default"],
      "outputs": ["{options.outputFile}"],
      "options": {
        "jestConfig": "{projectRoot}/jest.config.ts"
      }
    },
    "@nx/js:swc": {
      "dependsOn": ["^build"],
      "cache": true,
      "inputs": ["production", "^production"],
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/{projectRoot}",
        "main": "{projectRoot}/src/index.ts",
        "tsConfig": "{projectRoot}/tsconfig.lib.json",
        "assets": []
      }
    },
    "install": {
      "inputs": ["{projectRoot}/package.json"],
      "outputs": [
        "{projectRoot}/node_modules",
        "{projectRoot}/package-lock.json"
      ],
      "executor": "nx:run-commands",
      "options": {
        "command": "npm install",
        "cwd": "{projectRoot}"
      }
    },
    "ci": {
      "inputs": [
        "{projectRoot}/package.json",
        "{projectRoot}/package-lock.json"
      ],
      "outputs": ["{projectRoot}/node_modules"],
      "executor": "nx:run-commands",
      "options": {
        "command": "npm ci",
        "cwd": "{projectRoot}"
      }
    },
    "nx-release-publish": {
      "options": {
        "packageRoot": "dist/{projectRoot}"
      }
    }
  },
  "release": {
    "projects": ["*", "!source"],
    "projectsRelationship": "independent",
    "releaseTagPattern": "{projectName}/v{version}",
    "version": {
      "preVersionCommand": "npx nx run-many -t build",
      "generatorOptions": {
        "packageRoot": "{projectRoot}",
        "skipLockFileUpdate": false,
        "fallbackCurrentVersionResolver": "disk"
      },
      "conventionalCommits": true
    },
    "git": {
      "commitMessage": "chore: release [skip ci]"
    },
    "changelog": {
      "automaticFromRef": true,
      "projectChangelogs": true,
      "workspaceChangelog": false
    }
  },
  "tasksRunnerOptions": {
    "default": {
      "options": {
        "cacheableOperations": []
      }
    },
    "ci": {
      "runner": "@nx-aws-plugin/nx-aws-cache",
      "options": {
        "cacheableOperations": [],
        "awsRegion": "eu-central-1",
        "awsForcePathStyle": true
      }
    },
    "local": {
      "runner": "@nx-aws-plugin/nx-aws-cache",
      "options": {
        "cacheableOperations": [],
        "awsRegion": "eu-central-1",
        "awsForcePathStyle": true
      }
    }
  }
}
```

as for `project.json`, there isn't any `release` or `nx-release-publish` config found there.
Also according to the error message above, `nx/js` plugin should inject the `nx-release-publish` config to the project.json automatically 

`This is usually caused by not having an appropriate plugin, such as "@nx/js" installed, which will add the appropriate "nx-release-publish" target for you automatically.`

**Comment by or-orchid** [comment]
This code should handle it for `package.json` based libs. I can't find the equivalent for `project.json`

**Comment by JamesHenry** [comment]
@or-orchid are you saying that those libraries do not have a package.json file in their project root?

**Comment by or-orchid** [comment]
@JamesHenry 
This is our dir structure
```
.
├── README.md
├── commitlint.config.ts
├── jest.config.ts
├── jest.preset.js
├── libs
│   ├── redis
│   │   ├── CHANGELOG.md
│   │   ├── jest.config.ts
│   │   ├── package-lock.json
│   │   ├── package.json
│   │   ├── project.json
│   │   ├── src
│   │   ├── tsconfig.json
│   │   ├── tsconfig.lib.json
│   │   └── tsconfig.spec.json
├── migrations.json
├── nx.json
├── package-lock.json
├── package.json
├── tsconfig.base.json
├── tsconfig.build.json
└── tsconfig.json
```

redis/package.json
```
{
  "name": "@XXXXX/redis",
  "repository": {
    "type": "git",
    "url": "https://github.com/XXXXXX/YYYYY.git"
  },
  "version": "0.3.3",
  "type": "commonjs",
  "main": "./src/index.js",
  "typings": "./src/index.d.ts",
  "publishConfig": {
    "access": "restricted"
  },
  "peerDependencies": {
    "@nestjs/common": "^10.3.10",
    "@nestjs/config": "^3.1.1"
  },
  "dependencies": {
    "ioredis": "^5.4.0"
  },
  "devDependencies": {}
}
```

redis/project.json
```
{
  "name": "redis",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/redis/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nx/js:swc"
    },
    "tsc": {
      "executor": "@nx/js:tsc"
    },
    "lint": {
      "dependsOn": ["ci"],
      "executor": "@nx/eslint:lint"
    },
    "test": {
      "dependsOn": ["ci"],
      "executor": "@nx/jest:jest",
      "options": {
        "passWithNoTests": true
      }
    },
    "install": {},
    "ci": {}
  }
}
```

**Comment by or-orchid** [comment]
Bump

**Comment by JamesHenry** [comment]
@or-orchid Sorry for the delay, but as you can appreciate, things always go much faster if you take the time to provide a minimal reproduction repo for us to investigate.

I am now taking the time to manually recreate the outline you have listed

**Comment by JamesHenry** [comment]
What is the reason for having a nested lock file? Are you running a fully independent install command within that nested directory?

**Comment by JamesHenry** [comment]
Are you not using npm workspaces? These two points make a big difference here so I cannot continue further.

If you can provide a minimal reproduction repo that will naturally be most time efficient, but otherwise please clear guidance on how your package management concerns are set up.

**Comment by thdk** [comment]
@JamesHenry I'm currently working on making a demo repo to showcase some of my favorite Nx features to introduce Nx in our company. This error also shows up now. It's not a minimal reproduction, but it contains mainly bootstrapping two libraries and two applications to have some realistic projects to showcase.

Repo: https://github.com/thdk/nx-demo. 

```sh
# install
npm install
# start local verdaccio
npm run verdaccio 
# version and generate changelogs (I version both apps and libs with nx release)
npx nx release --skip-publish
# publish packages only
npx nx release publish --projects tag:type:package
```

I'll further investigate and also try on lower Nx version.

**Comment by thdk** [comment]
And a minimal repro: https://github.com/thdk/nx-repro-nx-release-publish-target-missing

Contains only a single project and configuration for `nx release` in `nx.json`.

```sh

Node           : 22.2.0
OS             : darwin-arm64
Native Target  : aarch64-macos
npm            : 10.8.0

nx                 : 19.6.5
@nx/js             : 19.6.5
@nx/linter         : 19.6.5
@nx/eslint         : 19.6.5
@nx/workspace      : 19.6.5
@nx/devkit         : 19.6.5
@nx/eslint-plugin  : 19.6.5
@nrwl/tao          : 19.6.5
typescript         : 5.5.4
---------------------------------------
Registered Plugins:
@nx/eslint/plugin

```

**Comment by JamesHenry** [comment]
@thdk Thank you for repro. In this case it’s because you have marked the library as private, which means it is not publishable, so there is correctly no associated target

https://github.com/thdk/nx-repro-nx-release-publish-target-missing/blob/51c841bd79c46230970c10b5adc6174abf1c46ec/some-lib/package.json#L10

**Comment by thdk** [comment]
@JamesHenry thanks for spotting the issue! 

I actually used the interactive approach to generate the library and I haven't been prompted to tell if the generate library should be publishable or not. My bad here. Do you think it would make sense to at that options to the interactive prompts? I could try to submit a PR for that :)

**Comment by tianbaolin** [comment]
I also encountered this issue and need help.

**Comment by or-orchid** [comment]
bump

**Comment by JamesHenry** [comment]
@or-orchid I don't really understand what you are bumping? I asked you a bunch of clarifying questions that you have not responded to, and you haven't provided a reproduction. A different community member provided a reproduction and I was immediately able to clarify that it was a setup issue. If their reproduction was accurate for your case too, then your issue has already been resolved. If not, then we are still waiting on your input. Please provide a minimal reproduction of the issue and help us to help you.

@tianbaolin please open a new issue with all the context around your specific case, as you have already seen, people often think they have the same issue as someone else but it commonly is not the case and that becomes clear when context is shared.

**Comment by JamesHenry** [comment]
@thdk Yes please feel free to submit a PR to improve the clarity of the publishable state of the library and tag me in it to review, thank you

**Comment by or-orchid** [comment]
Sorry, @JamesHenry , I was on vacation and just bumped, so that ticket won't be closed.
We are running a fully independent install command. Let me elaborate, and I'll work on a repro this weekend.
Our use case is a mono repo that contains **only** packages to be published and consumed by the different teams and projects in our organization.
Our packages are not configured as private in their `package.json` file.

Our different projects heavily rely on NX, specifically for `diff` calculation (affected) and releases. I would really prefer working with NX instead of NPM Workspaces.

I hope I can provide the repro ASAP

**Comment by or-orchid** [comment]
Just for future readers:
I had to add 
```
"nx-release-publish": {
      "options": {
        "packageRoot": "dist/{projectRoot}"
      }
    }
```
to each `project.json` file.
It wasn't required before 19.5.X, I think that there is something wrong with my workspace in general since it should be resolved automatically when using `@nx/js` plugin.

I start to think that this workspace configuration doesn't match my needs (a monorepo of independent packages only). I will start looking at npm workspaces
